### PR TITLE
Type Annotating: /steps/provision/connect.py

### PR DESCRIPTION
**Pull request by guy9050**
blocked because https://github.com/teemtee/tmt/pull/1247

### CODE REVIEW

File: tmt/steps/provision/connect.py [hunk 1]
```diff
@@ -61,21 +66,21 @@ def options(cls, how=None):
...
+    def default(self, option: str, default: Optional[Any] = None) -> Any:
         """ Return default data for given option """
         # User root as the default user
         if option == 'user':
             return 'root'
         # No other defaults available
         return default
 
-    def wake(self, keys=None, data=None):
+    def wake(self, keys: Optional[List[str]] = None, data: Any = None) -> None:
```
Comment (guy9050) [inline tmt/steps/provision/connect.py:77]: Is there better option for `data`?

Comment (thrix) [reply tmt/steps/provision/connect.py:77]: Yep, see: tmt/steps/prepare/ansible.py

Comment (guy9050) [reply tmt/steps/provision/connect.py]: Fixed

Comment (thrix) [reply tmt/steps/provision/connect.py]: it is not that simple, this plugin has different data than the one I referenced ...

**Comment by guy9050** [comment]
blocked because https://github.com/teemtee/tmt/pull/1247

**Comment by thrix** [comment]
yep, but let's MR ...

File: tmt/steps/provision/connect.py [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+StepDataType = TypedDict(
+    'StepDataType',
+    {
+        'playbook': List[str],
```
[the remark this answers is not in the document]
Comment (guy9050) [reply tmt/steps/provision/connect.py]: I dropped it since it's not necessary

**Comment by guy9050** [comment]
> yep, but let's MR ...

Updated. AFAIU, `provision` module is going to be changed drastically, so I'm waiting for the changes to land.

**Comment by thrix** [comment]
@guy9050 please rebase

**Review by thrix — APPROVED** [review]
LGTM

**Review by happz — APPROVED** [review]

**Pull request merged**